### ISSUE TITLE
Pulse audio dev headers missing causes unusual Linux startup problem

### ISSUE THREAD

**Issue by mattgodbolt** [tom-seddon/b2#36]
TL;DR - missing pulseaudio headers doesn't fail to build, but has unfortunate runtime issues.

--

Built from source (373050749ea0378a1589fbee896d8d8c81ec51b6) from the Linux instructions. Running the debug build pops up a "No such audio device" popup window and console output (after OKing) is:

```
~/d/b/b/d.linux (master|✔) 21.9s $ ./src/b2/b2
Gtk-Message: 21:25:55.968: Failed to load module "canberra-gtk-module"
b2 - BBC Micro B/B+/Master 128 emulator
failed to initialize audio: No such audio device
```

This is on Ubuntu 19.04.

A quick `grep` about makes it seem like `SDL`'s audio is causing the issue, that `SDL_OpenAudioDevice` is unable to find a sound output. Sadly this is where my knowledge dries up, so I turned to Google:

https://stackoverflow.com/questions/10465202/initializing-sdl-mixer-gives-error-no-available-audio-device

suggests I don't have a `/dev/dsp` and that means I was missing headers for pulse audio.

Indeed...that's the case, I must have missed them during initial install. `apt-get install libpulse-dev` and re-config...

and it works!

I guess this is a "please can the check for libpulse-dev fail if it's not present" feature request :)
